app/controllers/api_controller/vms.rb

```diff
@@ -1,5 +1,15 @@
 class ApiController
   module Vms
+    def vms_query_resource(object)
+      vm_attrs = attribute_selection_for("vms")
+      vm_attrs.blank? ? object.try(:vms) : Array(object.try(:vms)).collect do |vm|
+        add_hash = vm_attrs.each_with_object({}) do |attr, hash|
+          hash[attr] = vm.try(attr.to_sym) if vm.respond_to?(attr.to_sym)
```

Comment:
public_send actually makes better sense here, now that we know it responds to it.  I'll make the change.  Thanks !!